src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UsingStatement.cs

```diff
@@ -342,6 +366,8 @@ private BoundStatement RewriteUsingStatementTryFinally(SyntaxNode syntax, BoundB
             }
 
             // try { ... } finally { if (local != null) local.Dispose(); }
+            // or
+            // nullable or await variants
             BoundStatement tryFinally = new BoundTryStatement(
                 syntax: syntax,
```

Comment:
>syntax [](start = 24, length = 6)

I think it would be more appropriate to use corresponding using resource syntax for each individual BoundTryStatement, this way it will be easier to make sense what parts of tree related to what original source.

Reply:
This is probably applicable to all places where ```syntax``` is used in this function. So we probably should simply change what we passing in.

---
In reply to: [161342852](https://github.com/dotnet/roslyn/pull/23961#discussion_r161342852) [](ancestors = 161342852)

Reply:
Changing this breaks checks later in lowering, such as the one below.
I don't think it's worthwhile to change that existing code.

```C#
// in AsyncExceptionHandlerRewriter 
        public override BoundNode VisitTryStatement(BoundTryStatement node)
        {
            var tryStatementSyntax = node.Syntax;
            // If you add a syntax kind to the assertion below, please also ensure
            // that the scenario has been tested with Edit-and-Continue.
            Debug.Assert(
                tryStatementSyntax.IsKind(SyntaxKind.TryStatement) ||
                tryStatementSyntax.IsKind(SyntaxKind.UsingStatement) ||
                tryStatementSyntax.IsKind(SyntaxKind.ForEachStatement) ||
                tryStatementSyntax.IsKind(SyntaxKind.ForEachVariableStatement));
```